pkg/fetcher/fetcher.go

```diff
@@ -401,6 +406,19 @@ func (u *URLFetcher) IsStatusNotFound() bool {
 	return u.StatusCode == http.StatusNotFound
 }
 
+// IsNonretryableClientError returns true if status code is a nonretryable 4XX error. This includes
+// all 4XX errors except 'request timeout', 'locked', and 'too many requests'. Additionally, it
+// does not return 'not found' or 'unauthorized' errors since those are handled uniquely.
+func (u *URLFetcher) IsNonretryableClientError() bool {
```

Comment:
Would it be more readable/less error prone/and easy to modify if we just create a map of errors to bools and do a lookup here?